lib/ancestry/instance_methods.rb

```diff
@@ -288,16 +284,21 @@ def ancestry_callbacks_disabled?
 
   private
 
-    def cast_primary_key(key)
-      if [:string, :uuid, :text].include? primary_key_type
-        key
-      else
-        key.to_i
-      end
+    def parse_ancestry_column obj
+      cast_primary_keys(obj.to_s.split('/'))
+    end
+
+    def cast_primary_keys(keys)
+      skip_cast? ? keys : keys.map(&:to_i)
+    end
+
+    def skip_cast?
+      return @skip_cast if defined?(@skip_cast)
+      @skip_cast ||= [:string, :uuid, :text].include?(primary_key_type)
```

Comment:
You should make `[:string, :uuid, :text]` a constant to avoid allocating that every time this method is called (if you were doing this over multiple records, this would not be memoized every time).